packages/app/src/vis-packs/core/utils.ts

```diff
@@ -83,3 +89,11 @@ export function getSliceSelection(
 export function getImageInteractions(keepRatio: boolean) {
   return keepRatio ? BASE_INTERACTIONS : INTERACTIONS_WITH_AXIAL_ZOOM;
 }
+
+export function toNumArray(arr: ArrayValue<NumericLikeType>): NumArray {
+  if (typeof arr[0] === 'boolean') {
+    return arr.map((val) => (val ? 1 : 0));
+  }
+
+  return arr as NumArray;
```

Comment:
Scratched my head trying to remove this `as`, but two problems make this difficult:

- The value array is separated from the dataset object that describes the type, so narrowing the type of the dataset object does not help narrow the type of the value array...
- `ArrayValue<NumericType | BooleanType>` resolves to `(number | boolean)[] | TypedArray`. I actually [found a way](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types) to get `number[] | boolean[] | TypedArray` but it broke a bunch of code related to compound datasets ... and it makes sense for compound value arrays to have type `(number | string | ...)[] | TypedArray)`.

Reply:
Yeah. I have the feeling there is always a way to remove `as` but sometimes it is not worth the complexity.